src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,403 @@
+import FullScreenModalHeaderSubTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderSubTitle";
+import ToggleButton from "#SRC/js/components/ToggleButton";
+import HashMapDisplay from "#SRC/js/components/HashMapDisplay";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import Icon from "#SRC/js/components/Icon";
+import Util from "#SRC/js/utils/Util";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import RouterUtil from "#SRC/js/utils/RouterUtil";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+
+const METHODS_TO_BIND = [
+  "handleJSONToggle",
+  "handleGoBack",
+  "handleServiceReview",
+  "handleEditConfigurationButtonClick",
+  "handleActiveTabChange",
+  "handleFocusFieldChange",
+  "handleCloseConfirmModal",
+  "handleConfirmGoBack"
+];
+class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    const { activeTab, focusField } = this.getFirstTabAndField();
+
+    this.state = {
+      reviewActive: props.isInitialDeploy,
+      activeTab,
+      focusField,
+      jsonEditorActive: false,
+      isConfirmOpen: false,
+      isOpen: true
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleFocusFieldChange(activeTab, focusField) {
+    this.setState({ focusField, activeTab });
+  }
+
+  handleActiveTabChange(activeTab) {
+    const { packageDetails } = this.props;
+    const schema = packageDetails.getConfig();
+
+    const focusField = this.getFirstFieldInTab(schema, activeTab);
+
+    this.setState({ activeTab, focusField });
+  }
+
+  handleEditConfigurationButtonClick() {
+    const { activeTab, focusField } = this.getFirstTabAndField();
+
+    this.setState({ reviewActive: false, activeTab, focusField });
+  }
+
+  getHashMapRenderKeys(formData) {
+    if (!Util.isObject(formData)) {
+      return {};
+    }
+
+    let renderKeys = {};
+    Object.keys(formData).forEach(key => {
+      const formattedKey = key
+        .toLowerCase()
+        .split(/_|-/)
+        .map(word => StringUtil.capitalize(word))
+        .join(" ");
+
+      renderKeys[key] = formattedKey;
+      renderKeys = Object.assign(
+        renderKeys,
+        this.getHashMapRenderKeys(formData[key])
+      );
+    });
+
+    return renderKeys;
+  }
+
+  getFirstFieldInTab(schema, tab) {
+    // don't assume schema has two-levels of fields with properties
+    // for autofocus we return a field id name, if no fields return string "no-op"
+    if (
+      schema.properties[tab] &&
+      schema.properties[tab].properties &&
+      Object.keys(schema.properties[tab].properties).length >= 1
+    ) {
+      return Object.keys(schema.properties[tab].properties)[0];
+    }
+
+    return "NO-OP";
```

Comment:
I switch to use findNestedPropertyInObject, and now it will just return `null` which is fine. When it's null, it will simply not focus any tab, which is not an issue.